src/java.desktop/share/classes/javax/swing/text/DefaultEditorKit.java

```diff
@@ -642,6 +642,22 @@ public void write(Writer out, Document doc, int pos, int len)
      */
     public static final String endLineAction = "caret-end-line";
 
+    /**
+     * Name of the <code>Action</code> for moving the caret
+     * to the begin of a line it not at the begin already or
+     * upward one line otherwise.
+     * @see #getActions
+     */
```

Comment:
What ? We are adding new public API to solve this ? 
Why is that necessary ?

Reply:
Not public API..we are adding new editor "actions" if we need to keep previous fix..Without the previous fix JDK-8151136 , it works and we will not need to add these new actions.

Reply:
Actually, we do need the up/down new "action" even without JDK-8151136 fix...it was never working before..
We dont have action that will move the caret to begin/end of line and move down/up one line (if it's not at the begin/end).
It was either the former or the latter..